js/ElementRendererProvider.js

```diff
@@ -76,6 +80,96 @@ export default class ElementRendererProvider {
       return divEl;
     });
 
+    this.set(TYPES.SUBMIT, (config): HTMLElement => {
+      const divEl = document.createElement('div');
+      divEl.className = 'lp-json-pollock-element-submit-button';
+
+      if (config.rtl) {
+        divEl.dir = 'rtl';
+        Utils.addClass(divEl, 'direction-rtl');
+      }
+
+      const btnEl = document.createElement('button');
+      btnEl.innerHTML = Utils.normalizeHtmlText(config.title);
+
+      if (config.tooltip) {
+        btnEl.title = config.tooltip;
+        btnEl.setAttribute('aria-label', config.tooltip);
+      }
+      if (config.style) {
+        const style = Utils.styleToCss(config.style);
+        const splitedStyle = Utils.extractFromStyles(style, 'background-color');
+        divEl.setAttribute('style', splitedStyle.extractedStyle);
+        btnEl.style.cssText = splitedStyle.style;
+      }
+
+      if (config.click && config.click.actions) {
+        btnEl.onclick = this.wrapAction(config.click);
+      }
+
+      divEl.appendChild(btnEl);
+
+      return divEl;
+    });
+
+    this.set(TYPES.CHECKBOX, (config): HTMLElement => {
+      const divEl = document.createElement('div');
+      divEl.className = 'lp-json-pollock-element-checkbox';
+
+      if (config.rtl) {
+        divEl.dir = 'rtl';
+        Utils.addClass(divEl, 'direction-rtl');
+      }
+
+      const labelEl = document.createElement('label');
+      const checkEl = document.createElement('input');
+      const tmp = config;
+      checkEl.type = 'checkbox';
+      checkEl.name = config.name;
+      checkEl.value = config.value;
+      checkEl.className = 'lp-json-pollock-element-checkbox-input';
+      labelEl.appendChild(checkEl);
+
+      labelEl.className = 'lp-json-pollock-element-checkbox-label';
+      labelEl.innerHTML += Utils.normalizeHtmlText(config.text);
+
+      if (config.rtl) {
+        labelEl.dir = 'rtl';
+        Utils.addClass(labelEl, 'direction-rtl');
+      }
+      if (config.tooltip) {
+        labelEl.title = config.tooltip;
+        labelEl.setAttribute('aria-label', config.tooltip);
+      }
+      if (config.style) {
+        labelEl.style.cssText = Utils.styleToCss(config.style);
+      }
+      const borderEl = document.createElement('div');
+      borderEl.className = 'lp-json-pollock-border-element';
+      divEl.appendChild(borderEl);
+      divEl.appendChild(labelEl);
+
+      (divEl: any).afterRender = () => {
```

Comment:
add the `onclick` on `checkEl` instead of using `afterRender`

Reply:
actually I tried before and for some reason the attached event listener was not being triggered. However, when I added the event listener in `afterRender` function, it seemed to work. If you have any insights, please share with it with me and I will gladly make the change.